src/components/solverComponents/FileInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+interface IFileInputProps {
+    multiple?: boolean;
+    accept?: string;
+    onFileChanged?: (files: FileList | null) => void;
+}
+
+interface IFileInputState {
+    inputRef: React.MutableRefObject<HTMLInputElement | null>;
+}
+
+export class FileInput extends React.Component<IFileInputProps, IFileInputState> {
+    constructor(props: IFileInputProps) {
+        super(props);
+
+        this.state = {
+            inputRef: React.createRef()
+        }
+    }
+
+    openInput(): void {
+        this.state.inputRef.current?.click()
+    }
+
+    render() {
+        return (
+            <label>
```

Comment:
labels usually only contain text. Why is there a label anyways?

Reply:
@Elscrux explain yourself

Reply:
That can be deleted. I added it when I tried a different approach first.

Reply:
deleted